fastlane/lib/fastlane/actions/setup_circle_ci.rb

```diff
@@ -24,15 +24,20 @@ def self.setup_output_paths(params)
         ENV["SCAN_INCLUDE_SIMULATOR_LOGS"] = true.to_s
       end
 
-      def self.setup_keychain
+      def self.setup_keychain(params)
```

Comment:
I'd suggest passing in either the password or the random password flag as parameter instead of the params hash for increased readability